### PR TITLE
Tool resprite

### PR DESCRIPTION
**Pull request by Bm0n**
## What Does This PR Do

Updates almost every tool sprite to a newer more cooler version, basically a port of https://github.com/tgstation/tgstation/pull/58616 which was a hugely tweaked version of a Bee PR which retain the SOUL of the tool sprites (i am not kidding, most of the tools just reshades with better colours)

RCD and RPD from https://github.com/Citadel-Station-13/Citadel-Station-13/pull/8953

RCD now has a build and deconstruct animation, port of https://github.com/tgstation/tgstation/pull/47894

Few other tweaks and additions for Para specifically by me

Fixes https://github.com/ParadiseSS13/Paradise/issues/19864

the rcd wall deconstruct timer was extended by 1 second so I is don't have to make a snowflake animation for a 1 second difference

## Why It's Good For The Game

hey... remember when recently we got all those new surgical tool sprites from that two letter community that everyone flips out about when they hear its name uttered along with the words "sprite pr"? think that but now with actual tools.

In all seriousness, our tools are crusty and it shows, out of any server I have played on we have, hands down, the worst tool sprites overall, I am going to fix that now. I took the safe route by porting TG's tools which are mostly just a modernized version of our old tools.

## Images of changes

i swear to god if you judge all of these sprites based on this one SOMEWHAT BLURRY IMAGE I will forever own your soul. Go look at the .DMI file, stop being lazy, also there's more than just this.
image

The RCD building animation.

https://streamable.com/9416f8

## Testing

Yes, in game.
## Changelog
:cl: Bmon, MrDoomBringer, Michiyamenotehifunana, maxymax13, PositiveEntropy, Kryson, Flareguy
add: The RCD now has a building animation.
imageadd: Almost every tool has gotten a new sprites!
fix: The RCD now spark again when switching modes in the radical menu.
/:cl:

### CODE REVIEW

**Comment by Bm0n** [comment]
fixed TGs sad excuse of a medical wrench
image

also the new welder fuel gauge has been made consistent
image

**Comment by McRamon** [comment]
not a big fan of some of them, like jaws or multitool are noisy  and jarring. wrench is awesome tho.

**Comment by Denthamos** [comment]
Agreeing with ramon.  The mulitool is somewhat passible, the jaws however are a bit much there.  The rest of the tools however, very nice and I love the RCD animation.

**Comment by Bm0n** [comment]
> Agreeing with ramon. The mulitool is somewhat passible, the jaws however are a bit much there. The rest of the tools however, very nice and I love the RCD animation.

unless either of you can give more specific criticisms about it I cannot improve it

**Comment by McRamon** [comment]
> unless either of you can give more specific criticisms about it I cannot improve it

Crowbar, screwdriver, rcd - no complaints
Wrench - minor thing for medical wrench - due to the tone of blue cross and outer grey part of the wrench handle, it really jsut blends together. I look at it and i see just a dirt spot in the middle. Ofc not gamebreaking ,but kinda would be cool if the blue cross was distinct and clear.
RPD - generally its ok, kinda feels like it night be overcomplicated a bit but nothing serious
Drill - nice and clean, however the shape is not perfect and kinda a bit broken at times. Not a major concern, just a nitpick
Jaws - too noisy, doesnt go well with other tools. Also not a huge problem, but aint a perfect sprite.
Multitool - Kinda weird, basically same as current multitool but with some gradient shading which doesnt look really good, especially on the red one.
Welders - the weirdes one here, they are kinda changed, but the same? Can't even say the look anyhow different, they remain as old outdated sprites with different shading, which isn't really that much better, they need a complete redesign.

If you need any help on how to improve any of it, just @me on disc

**Comment by Bm0n** [comment]
thank you for the feedback

>Wrench - minor thing for medical wrench - due to the tone of blue cross and outer grey part of the wrench handle, it really jsut blends together. I look at it and i see just a dirt spot in the middle. Ofc not gamebreaking ,but kinda would be cool if the blue cross was distinct and clear. 

That should be an easy thing to do

>RPD - generally its ok, kinda feels like it night be overcomplicated a bit but nothing serious 

Probably not going to change the RPD, I think it needs to be bigish since it shoots out pipes and I can't think of any other way to change it.

>Drill - nice and clean, however the shape is not perfect and kinda a bit broken at times. Not a major concern, just a nitpick 

This is one of the sprites I touched up from TG since I got a few complaints about it in the spriter channel. The 3/4ing on [the old version](https://i.imgur.com/mEvuZih.png) was kind of messed up so I fixed it by extending the bottom half downwards to match the 3/4ing on the top. I think where I butchered it was around the bottom right-hand side of the base, I am going to try to flatten it out and see how that works.

>Jaws - too noisy, doesnt go well with other tools. Also not a huge problem, but aint a perfect sprite. 

I think the reason it looks this way is that the sprite had to be recoloured to appease TGs soul crowd. [The original bee version](https://i.imgur.com/DbpugCV.png) looked like this and I think has more noticeable shape to it compared to the TG version.

But I'll see what I can do about the jaws, I am not the best shader but I think I can remove a lot of the noise from it.


>Multitool - Kinda weird, basically same as current multitool but with some gradient shading which doesnt look really good, especially on the red one. 

Again, something done to appease the muh soul people on TG, if you put [both side by side ](https://i.imgur.com/05YW8qc.png)you can tell its literally just a reshade with a different lighting angle and better colours IMO.

I don't know what can be done to improve it other than making it flatter with its shading. The red ones colours are kind of scuffed, I'll try to match it more to the yellow one in terms of brightness.

>Welders - the weirdes one here, they are kinda changed, but the same? Can't even say the look anyhow different, they remain as old outdated sprites with different shading, which isn't really that much better, they need a complete redesign.

Yeeeah, again much like the multitool, TG didn't like the Bee tools so they ended up just reshading the old ones to be more consistent and less messy, [side by side](https://i.imgur.com/4ncTRYN.png) again. My honest opinion is that if it can't get merged on TG because people are bickering about soul then there is no way in hell it'd be merged on Paradise, we tend to have more muh soul type people here.

Just for reference though here's all of the [original sprites from Bee](https://user-images.githubusercontent.com/53237389/93640092-61f23b00-f9d0-11ea-9bf5-1bcfc560d5c8.png) just so you can look at them.


> If you need any help on how to improve any of it, just @me on disc

Will do

File: code/game/objects/items/weapons/RCD.dm [hunk 1]
```diff
@@ -563,12 +581,16 @@
...
 		if(MODE_AIRLOCK)
 			. = mode_airlock(A, user)
+			update_icon()
 		if(MODE_DECON)
 			. = mode_decon(A, user)
+			update_icon()
 		if(MODE_WINDOW)
 			. = mode_window(A, user)
+			update_icon()
 		else
```
Comment (S34NW) [inline code/game/objects/items/weapons/RCD.dm:594]: These can be just once following the switch

File: code/game/objects/items/weapons/RCD.dm [hunk 2]
```diff
@@ -597,7 +619,18 @@
...
+		to_chat(user, no_ammo_message)
+		flick("[icon_state]_empty", src)
+
+	return .
+
+/obj/item/rcd/update_icon()
+	..()
+	var/ratio = CEILING((matter / max_matter) * 10, 1)
+	cut_overlays()
+	add_overlay("[icon_state]_charge[ratio]")
```
Comment (S34NW) [inline code/game/objects/items/weapons/RCD.dm:634]: We tend to split these into their various sub-procs these days (`update_overlays()` in this case), I think the same can be done here. Then you neeed to update all of your `update_icon()` calls with `update_icon(UPDATE_OVERLAYS)`

File: code/game/objects/items/weapons/RCD.dm [hunk 3]
```diff
@@ -597,7 +619,18 @@
  * * amount - an amount of matter to check for
  */
 /obj/item/rcd/proc/checkResource(amount, mob/user)
-	return matter >= amount
+	. = matter >= amount
+	if(!. && user)
+		to_chat(user, no_ammo_message)
+		flick("[icon_state]_empty", src)
+
+	return .
```
Comment (S34NW) [inline code/game/objects/items/weapons/RCD.dm:627]: you don't need to specify the `.` here, IIRC the default return value is `.`

**Review by S34NW — COMMENTED** [review]
Some changes

File: code/game/objects/items/weapons/RCD.dm (hunk 2 above)
Comment (Bm0n) [reply code/game/objects/items/weapons/RCD.dm:634]: yes this works, I'll do this.

Trying to wrap my head around this though. What is the actual meaningful difference between `update_overlays()` and `update_icon()` if they both do the exact same thing

File: code/game/objects/items/weapons/RCD.dm (hunk 1 above)
Comment (Bm0n) [reply code/game/objects/items/weapons/RCD.dm:594]: i have monkey brain. this is fixed

File: code/game/objects/items/weapons/RCD.dm (hunk 3 above)
Comment (Bm0n) [reply code/game/objects/items/weapons/RCD.dm:627]: didn't know it was literally the same, lmao

File: code/game/objects/items/weapons/RCD.dm (hunk 2 above)
Comment (S34NW) [reply code/game/objects/items/weapons/RCD.dm]: It's going to affect you less here, but it means we can have granular control over what we want to update, saving CPU. No need to rebuild an entire human icon when you just want to remove a blood overlay, for example.

**Comment by Bm0n** [comment]
Apologies, will work on this in the coming days, was busy with another PR.

**Comment by Bm0n** [comment]
New changes:

- Screwdriver is now pointed
- Jaws of life resprited to be a lot cleaner, includes inhands and belt slot icon
- The hand drills base has been fixed around the base of the sprite
- Using multool from CEV Eris, Ramon suggested this.
- Medical wrench tweaked so that the cross is more visible

image

I didn't touch the welders because I couldn't find anything better. Really these new TG ones are just more or less better-shaded versions of our old ones. If anyone else wants to PR new welders please do so.

File: code/game/objects/items/weapons/RCD.dm [hunk 4]
```diff
@@ -451,59 +463,63 @@
...
-			if(do_after(user, 40 * toolspeed, target = A))
+			var/obj/effect/temp_visual/rcd_effect/reverse/E = new(get_turf(A))
+			if(do_after(user, 50 * toolspeed, target = A))
 				if(!useResource(5, user))
 					return FALSE
 				playsound(loc, usesound, 50, 1)
 				var/turf/AT = A
 				AT.ChangeTurf(/turf/simulated/floor/plating)
 				return TRUE
+			qdel(E)
```
Comment (S34NW) [inline code/game/objects/items/weapons/RCD.dm:474]: I'm worried that these deletions of the visual effects can not be reached under some conditions, such as onces creating returns above.

Have you verified the effect properly cleans up under fail scenarios?

Comment (S34NW) [reply code/game/objects/items/weapons/RCD.dm:474]: There are a number of them like this, FYI

**Comment by jughu** [comment]
> fixed TGs sad excuse of a medical wrench image
> 
> also the new welders fuel gauge has been made consistent image

it wasnt a excuse it was a meme because the penlite ontop of the wrench

File: code/game/objects/items/weapons/RCD.dm (hunk 4 above)
Comment (Bm0n) [reply code/game/objects/items/weapons/RCD.dm:474]: shouldn't be any situation where it wouldn't qdel the animation at least that I can think of.

also yes you can see an example in the video linked in the PR

**Review by S34NW — APPROVED** [review]
Code looks fine, icons appear okay from looking at IDB

File: code/game/objects/effects/temporary_visuals/miscellaneous.dm [hunk 5]
```diff
@@ -363,3 +363,25 @@
...
+	icon_state = "rcd_end"
+	duration = 11
+
+/obj/effect/temp_visual/rcd_effect/reverse
+	icon_state = "rcd_reverse"
+	duration = 61
+
+/obj/effect/temp_visual/rcd_effect/reverse_short
+	icon_state = "rcd_short_reverse"
+	duration = 31
```
Comment (hal9000PR) [inline code/game/objects/effects/temporary_visuals/miscellaneous.dm:387]: would you make these durations use the seconds define

File: code/game/objects/items/weapons/RCD.dm [hunk 6]
```diff
@@ -451,59 +463,63 @@
 		if(checkResource(5, user))
 			to_chat(user, "Deconstructing Wall...")
 			playsound(loc, 'sound/machines/click.ogg', 50, 1)
-			if(do_after(user, 40 * toolspeed, target = A))
+			var/obj/effect/temp_visual/rcd_effect/reverse/E = new(get_turf(A))
+			if(do_after(user, 50 * toolspeed, target = A))
```
Comment (hal9000PR) [inline code/game/objects/items/weapons/RCD.dm:467]: seconds define on this delay aswell please

**Review by hal9000PR — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/RCD.dm (hunk 6 above)
Comment (Bm0n) [reply code/game/objects/items/weapons/RCD.dm]: should all be using seconds now

File: code/game/objects/effects/temporary_visuals/miscellaneous.dm (hunk 5 above)
Comment (Bm0n) [reply code/game/objects/effects/temporary_visuals/miscellaneous.dm]: sure

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**